source/core/TimeFunctions.cpp

```diff
@@ -166,11 +166,14 @@ namespace Vireo {
     }
 
     //------------------------------------------------------------
-    VIREO_FUNCTION_SIGNATURE2(DateTimeToTimestamp, LVDateTimeRec, Timestamp)
+    VIREO_FUNCTION_SIGNATURE3(DateTimeToTimestamp, LVDateTimeRec, Boolean, Timestamp)
     {
         LVDateTimeRec *dt = _ParamPointer(0);
+        Boolean isUTC = _ParamPointer(1) ? _Param(1) : false;
+        Int32 timeZoneOffset = isUTC ? 0 : Date::getLocaletimeZone();
         Timestamp timestamp(dt->fractional_secs, dt->second, dt->minute, dt->hour, dt->day_of_month, dt->month, dt->year);
-        _Param(1) = timestamp;
+        Timestamp local = timestamp - timeZoneOffset;
```

Comment:
`Timestamp local = timestamp - timeZoneOffset;`
This is misleading/confusing to follow the code (and can possibly be slightly incorrect). We should subtract only Timestamp._integer part and Timestamp._fraction should not be touched. It is unfortunate that there is an operator+ overload for an Int64 argument but there is not an operator- overload for an Int64 argument. Instead of adding a new operator- overload with Int64 argument to fix this and make the code clearer,  I think we should remove the existing operator+(Int64) overload and add a new method in Date class (called UpdateTimeZoneOffset) and call that.